src/App.tsx

```diff
@@ -1,93 +1,147 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { UserWarning } from './UserWarning';
+import {
+  createTodo,
+  deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+
+const USER_ID = 7075;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isEnterPressed, setIsEnterPressed] = useState(false);
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [allTodosCompleted, setAllTodosCompleted] = useState<boolean>(true);
+  const count = todos.filter(todo => !todo.completed).length;
+
+  useEffect(() => {
+    getTodos(USER_ID).then(todosFromServer => {
+      setTodos(todosFromServer);
+    });
+  }, [isEnterPressed,
+  ]);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo: Todo) => {
+      switch (selectFilter) {
+        case 'active':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return todos;
+      }
```

Comment:
No need to filter if `selectFilter` is equal `all`.
NOTE: filter callback should return `boolean` value - your case you return an `array` in the `default` case